context/VersionContext.js

```diff
@@ -0,0 +1,105 @@
+import { createContext, useContext, useState, useEffect } from "react";
+import getConfig from "next/config";
+const { publicRuntimeConfig } = getConfig();
+
+const VersionContext = createContext();
+
+const DEFAULT_VERSION = publicRuntimeConfig.DEFAULT_VERSION;
+const DEFAULT_PLATFORM = publicRuntimeConfig.DEFAULT_PLATFORM;
+const VERSIONS_LIST = publicRuntimeConfig.VERSIONS_LIST;
+const LATEST_VERSION = publicRuntimeConfig.LATEST_VERSION;
+const PLATFORM_VERSIONS = publicRuntimeConfig.PLATFORM_VERSIONS;
+const PLATFORM_LATEST_VERSIONS = publicRuntimeConfig.PLATFORM_LATEST_VERSIONS;
+const PLATFORMS = publicRuntimeConfig.PLATFORMS;
+
+export function VersionContextProvider(props) {
+  const [platform, setPlatform] = useState(
+    () => props.platformFromSlug ?? DEFAULT_PLATFORM, // Init: Platform from slug, else "oss"
+  );
+  const [version, setVersion] = useState(
+    () => props.versionFromSlug ?? DEFAULT_VERSION, // Init: Version from slug, else "latest"
+  );
+
+  useEffect(() => {
+    if (props.versionFromSlug || props.platformFromSlug) {
+      setVersion(props.versionFromSlug);
+      setPlatform(props.platformFromSlug);
+    }
+  }, [props.versionFromSlug, props.platformFromSlug]);
+
+  function goToLatest() {
+    setVersion(DEFAULT_VERSION);
+  }
```

Comment:
I think these functions need be declared with useCallback to avoid creating a new instance every time:

```
  const goToLatest = useCallback(() => {
    setVersion(DEFAULT_VERSION);
  }, []);

  const goToOpenSource = useCallback(() => {
    setPlatform(DEFAULT_PLATFORM);
  }, []);
```